website/blog/2018-04-04-using-typescript-with-react-native.md

```diff
@@ -0,0 +1,310 @@
+---
+title: Using TypeScript with React Native
+author: Ash Furrow
+authorTitle: Software Engineer at Artsy
+authorURL: https://github.com/ashfurrow
+authorImage: https://avatars2.githubusercontent.com/u/498212?s=460&v=4
+authorTwitter: ashfurrow
+category: engineering
+---
+
+JavaScript! We all love it. But someone us also love [types](https://en.wikipedia.org/wiki/Data_type). Luckily, options exist to add stronger types to JavaScript. My favourite is [TypeScript](https://www.typescriptlang.org), but React Native supports [Flow](https://flow.org) out of the box. Which you prefer is a matter of preference, they each have their own approach on how to add the magic of types to JavaScript. Today, we're going to look at how to use TypeScript in React Native apps.
+
+We'll be using Microsoft's [TypeScript-React-Native-Starter](https://github.com/Microsoft/TypeScript-React-Native-Starter) repo as a guide.
+
+## Prerequisites
+
+Because you might be developing on one of several different platforms, targeting several different types of devices, basic setup can be involved. You should first ensure that you can run a plain React Native app without TypeScript. Follow [the instructions on the React Native website to get started](https://facebook.github.io/react-native/docs/getting-started.html). When you've managed to deploy to a device or emulator, you'll be ready to start a TypeScript React Native app.
+
+You will also need [Node.js](https://nodejs.org/en/), [NPM](https://www.npmjs.com), and [Yarn](https://yarnpkg.com/lang/en).
+
+## Initializing
+
+Once you've tried scaffolding out an ordinary React Native project, you'll be ready to start adding TypeScript. Let's go ahead and do that.
+
+```sh
+react-native init MyAwesomeProject
+cd MyAwesomeProject
+```
+
+## Adding TypeScript
+
+The next step is to add TypeScript to your project. The following commands will:
+
+* add TypeScript to your project
+* add [React Native TypeScript Transformer](https://github.com/ds300/react-native-typescript-transformer) to your project
+* initialize an empty TypeScript config file, which we'll configure next
+* add an empty React Native TypeScript Transformer config file, which we'll configure next
+* adds [typings](https://github.com/DefinitelyTyped/DefinitelyTyped) for React and React Native
+
+Okay let's go ahead and run these.
+
+```sh
+yarn add --dev typescript
+yarn add --dev react-native-typescript-transformer
+yarn tsc --init --pretty --jsx react
+touch rn-cli.config.js
+yarn add --dev @types/react @types/react-native
+```
+
+The `tsconfig.json` file contains all the settings for the TypeScript compile. The defaults created by the command above are mostly fine, but open the file and uncomment the following line:
+
+```js
+{
+  ...
```

Comment:
I'm not a fan of using "..." here. It can be confusing for someone new to JavaScript to see the `...` spread operator after getting used to seeing  "..." to convey truncation in code examples.

Reply:
Great point! I'll replace it with a comment explaining this is only part of a larger snippet.